pontoon/base/models.py

```diff
@@ -2025,29 +2025,30 @@ def save(self, imported=False, *args, **kwargs):
             if self.approved:
                 self.entity.mark_changed(self.locale)
 
-            # Check and update the latest translation where necessary.
-            self.check_latest_translation(self.entity.resource.project)
-            self.check_latest_translation(self.locale)
-            self.check_latest_translation(translatedresource)
-
-            project_locale = utils.get_object_or_none(
-                ProjectLocale,
-                project=self.entity.resource.project,
-                locale=self.locale
-            )
-            if project_locale:
-                self.check_latest_translation(project_locale)
+            # Update latest translation where necessary
+            self.update_latest_translation()
 
-    def check_latest_translation(self, instance):
+    def update_latest_translation(self):
         """
-        Check if the given model instance has a `latest_activity`
-        attribute and, if it does, see if this translation is more
-        recent than it. If so, replace it and save.
+        Set `latest_translation` to this translation if its more recent than
+        the currently stored translation. Do this for all affected models.
         """
-        latest = instance.latest_translation
-        if latest is None or self.latest_activity['date'] > latest.latest_activity['date']:
-            instance.latest_translation = self
-            instance.save(update_fields=['latest_translation'])
+        resource = self.entity.resource
```

Comment:
Just curious - did you (as part of this PR) also review the entire #601, or do you still want to do that separately?